src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -496,6 +496,8 @@ private static ScriptType ParseScriptType(string scriptFilePath)
 
         internal Collection<FunctionDescriptor> ReadFunctions(List<FunctionMetadata> metadatas, IEnumerable<FunctionDescriptorProvider> descriptorProviders)
         {
+            metadatas = ApplyMaximumFunctionCount(metadatas);
```

Comment:
Thought about that, but wanted to keep that logic isolated and working on top of the parsed metadata.
I think you're right, though. Although this is super simple as is, we can probably have something simpler and avoid some unnecessary processing by adding this check to the loop.